PowerShell/JumpCloud Commands Gallery/Windows Commands/Windows - Install Remote Assist.md

```diff
@@ -0,0 +1,81 @@
+#### Name
+
+Windows - Install Remote Assist | v1.0 JCCG
+
+#### commandType
+
+windows
+
+#### Command
+
+```
+$installerURL="https://cdn.awsstg.jumpcloud.com/TheJumpCloud/jumpcloud-remote-assist-agent/latest/jumpcloud-assist-app.exe"
+$JumpCloudThumbprint="7A4844FBF481047BEDBB7A8054069C50E449D355"
+$installerTempLocation="C:\Windows\Temp\JumpCloudRemoteAssistInstaller.exe"
+
+Write-Host "Downloading JumpCloud Remote Assist installer now."
+try {
+    $ProgressPreference = "SilentlyContinue"
+    Invoke-WebRequest -Uri $installerURL -OutFile $installerTempLocation
+}
+catch {
+    Write-Error "Unable to download JumpCloud Remote Assist installer."
+    exit 1
+}
+Write-Host "Finished downloading JumpCloud Remote Assist installer."
+
+try {
+    Write-Host "Verifying Authenticode Signature"
+    $authenticode = Get-AuthenticodeSignature "$installerTempLocation"
+    if ( $authenticode.Status -ne "Valid" )
+    {
+        Write-Error "No valid Authenticode signature found, aborting installation"
+        exit 1
+    }
+
+    if ( $authenticode.SignerCertificate.Thumbprint -ne $JumpCloudThumbprint )
+    {
+        Write-Error "No valid Authenticode signature found, aborting installation"
+        exit 1
+    }
+
+    try {
+        Get-Process "JumpCloud Assist App" -ErrorAction Stop | Stop-Process
+    }
+    catch [Microsoft.PowerShell.Commands.ProcessCommandException] {
+        # Not a problem if the app isn't already running
+    }
+    catch {
+        Write-Error "Failed to stop the running JumpCloud Remote Assist App."
+        exit 1
+    }
+
+    Write-Host "Installing JumpCloud Remote Assist now, this may take a few minutes."
+    try {
+        $installerProcess = Start-Process -FilePath $installerTempLocation -Wait -PassThru
+    }
+    catch {
+        Write-Error "Failed to run JumpCloud Remote Assist installer."
+        exit 1
+    }
+
+    Write-Host "JumpCloud Remote Assist installer returned $($installerProcess.ExitCode)."
+
+    exit $installerProcess.ExitCode
+}
+finally {
+    Remove-Item "$installerTempLocation"
```

Comment:
I don't think we should be removing the $installerTempLocation this is causing an error with the shortcut to the binary in the desktop

Reply:
I'm not seeing the connection, but this is the sort of thing I struggle with on Windows: "Action A has no obvious causal relationship to result B." I never now how to respond :)

Reply:
I'm not seeing the connection either :/ I just noticed that removing the temporary file is resulting in not being able to open the binary ; see attached 

![ras1](https://user-images.githubusercontent.com/104375988/182678354-f3f685e0-0f6c-4e91-b003-6a29879c3053.png)
![ras2](https://user-images.githubusercontent.com/104375988/182678367-f8c1ed2e-bfa6-48cb-a679-672ff9ceefa9.png)

And if the script is updated with `Remove-Item "$installerTempLocation"` removed

![ras_desktop1](https://user-images.githubusercontent.com/104375988/182678534-6249966a-4781-443c-8eee-a1012313c8c5.png)
![ras_desktop2](https://user-images.githubusercontent.com/104375988/182678540-39392482-3c17-497a-aeee-10533ab90af4.png)

fwiw, we do have other files that are JC installed which live in `C:\Windows\Temp`

Reply:
Looks like the problem has to do with whether or not the user double clicking on the icon is a "regular user" or "has admin privileges" (not 100% sure the correct terminology on Windows). Since the Remote Assist Agent EXE is in `C:\Program Files\Jumpcloud`, and that folder is correctly restricted to admin-read only, a non-admin cannot and should not be able to access the files in that folder. The current plan is to change the installed location to one where the permissions can be changed such that all users can read and execute the contents.